File: LinkedList/KosaKosa LinkedList/Code.js
import java.io.*;
import java.util.*;
import java.text.*;
import java.math.*;
import java.util.regex.*;

class Node{
    int data;
    Node next;
    
    Node(int d){
        this.data = d;
        this.next = null;
    }
}

public class Solution {
    
    static Node head = null;
    static Node tail = null;
    
    static void insert(int d){
        Node nn = new Node(d);
        if( head == null){ 
            head = nn; 
            tail = nn; 
        }
        else{ 
            tail.next = nn; 
            tail = nn; 
        }
    }
    
    static void dis(){
        Node temp = head;
        while(temp.next != null){
            System.out.print(temp.data+" ");
            temp = temp.next;
        }
        System.out.print(temp.data+" ");
    }
    
    static void mid(){
        Node slow = head;
        Node fast = head;
        
        while(fast !=null && fast.next !=null ){
            slow = slow.next;
            fast = fast.next.next;
        }
        System.out.print(slow.data);        
    }
    
    

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        List<Integer> li = new ArrayList<>();
        while(sc.hasNext()){
            int d = sc.nextInt();
            if(!li.contains(d)){
                li.add(d);
                insert(d);
            }
        }
        dis();
        System.out.println();
        mid();
        sc.close();
    }
}

File: Graph/Print adjacency list 3/code .java
import java.io.*;
import java.util.*;

public class Solution {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        
        for(int time = 0; time<t; time++){
            
            int v = sc.nextInt();
            int e = sc.nextInt();
            
            List<List<Integer>> li = new ArrayList<>();
            for(int i=0; i<v; i++){
                li.add(new ArrayList<>());
            }
            
            for(int i=0; i<e; i++){
                int st = sc.nextInt();
                int ev = sc.nextInt();
                li.get(st).add(ev);
                li.get(ev).add(st);
            }
        
            for(int i=0; i<li.size(); i++){
                System.out.print(i);
                for(int j=0; j<li.get(i).size(); j++){
                    System.out.print("-> "+li.get(i).get(j));
                }
                System.out.println();
            }
        }
    }
}
